src/globus_sdk/services/compute/data.py

```diff
@@ -0,0 +1,47 @@
+import typing as t
+
+from globus_sdk import utils
+from globus_sdk._types import UUIDLike
+
+
+class FunctionMetadata(utils.PayloadWrapper):
+    """A wrapper for function metadata.
+
+    :param python_version: The Python version used to serialize the function.
+    :param sdk_version: The Globus Compute SDK version used to serialize the function.
+    """
+
+    def __init__(self, *, python_version: str, sdk_version: str):
+        super().__init__()
+        self["python_version"] = python_version
+        self["sdk_version"] = sdk_version
```

Comment:
Can we define defaults for these fields from the current execution environment?

The former looks like it's formatted `sys.version_info` data. I just tried out this little ditty and it seems to do the right thing:
```python
import sys


def fmt_version(v) -> str:
    if v.releaselevel == "final":
        return f"{v.major}.{v.minor}.{v.micro}"
    else:
        designator = {
            "candidate": "rc",
            "beta": "b",
            "alpha": "a",
        }[v.releaselevel]
        return f"{v.major}.{v.minor}.{v.micro}{designator}{v.serial}"


print(fmt_version(sys.version_info))
```

We could do something similar to pull the `globus-compute-sdk` version using importlib metadata.

---

This is as much a question about understanding of the service as it is about what we could do in this class.
When are callers expected to construct this object? Is it typically referring to the current runtime environment or are there significant cases in which it is meant to refer to some _other_ runtime when data was constructed?

Reply:
It refers to the runtime environment when serializing the function. We currently set these values in the Compute SDK, where the serialization takes place. I prefer to leave it as-is for now.

Reply:
Okay. It's easier to add defaults than it is to take them away (the former is backwards compatible but the latter is not), so I see no reason to block on this. Mostly was just curious. :slightly_smiling_face: